### PR TITLE
Add xml subnet docs

### PR DESCRIPTION
**Pull request by bquantump**
https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_boards/board/t/DotNET%20Management%20SDK%20Team/Stories/?workitem=5053

### CODE REVIEW

File: azure-proto-network/SubnetContainer.cs [hunk 1]
```diff
@@ -41,26 +53,36 @@ public async override Task<ArmResponse<Subnet>> CreateAsync(string name, SubnetD
...
+        /// <inheritdoc/>
         public override ArmOperation<Subnet> StartCreate(string name, SubnetData resourceDetails, CancellationToken cancellationToken = default)
         {
             return new PhArmOperation<Subnet, Azure.ResourceManager.Network.Models.Subnet>(
                 Operations.StartCreateOrUpdate(Id.ResourceGroup, Id.Name, name, resourceDetails.Model, cancellationToken),
                 s => new Subnet(ClientOptions, new SubnetData(s, Location.Default)));
         }
 
+        /// <inheritdoc/>
         public async override Task<ArmOperation<Subnet>> StartCreateAsync(string name, SubnetData resourceDetails, CancellationToken cancellationToken = default)
```
Comment (allenjzhang) [inline azure-proto-network/SubnetContainer.cs:65]: nit:  ->  public override async

**Comment by allenjzhang** [comment]
Are you missing Subnet.cs and SubnetData.cs?

File: azure-proto-network/Placeholder/SubnetData.cs [hunk 2]
```diff
@@ -5,77 +5,118 @@
...
     {
         public SubnetData(Azure.ResourceManager.Network.Models.Subnet sub, string location) : base(sub.Id, sub)
         {
         }
 
+        /// <summary>
+        /// Gets the subnet id.
+        /// </summary>
         public override string Name => Model.Name;
+         /// <summary> The provisioning state of the subnet resource. </summary>
```
Comment (m-nash) [inline azure-proto-network/Placeholder/SubnetData.cs:20]: missing newline

File: azure-proto-network/Placeholder/SubnetData.cs [hunk 3]
```diff
@@ -5,77 +5,118 @@
...
         public string Purpose => Model.Purpose;
+        
+        /// <summary> An array of references to the delegations on the subnet. </summary>
         public IList<Delegation> Delegations
         {
             get => Model.Delegations;
             set => Model.Delegations = value;
         }
+
+        /// <summary>An array of references to services injecting into this subnet. </summary>
```
Comment (m-nash) [inline azure-proto-network/Placeholder/SubnetData.cs:36]: missing space

File: azure-proto-network/Placeholder/SubnetData.cs [hunk 4]
```diff
@@ -5,77 +5,118 @@
...
         public IList<ServiceAssociationLink> ServiceAssociationLinks => Model.ServiceAssociationLinks;
+
+        /// <summary>An array of references to the external resources using subnet. </summary>
```
Comment (m-nash) [inline azure-proto-network/Placeholder/SubnetData.cs:39]: missing space

File: azure-proto-network/Placeholder/SubnetData.cs [hunk 5]
```diff
@@ -5,77 +5,118 @@
...
 
+        /// <summary>
+        /// Gets the subnet id.
+        /// </summary>
         public override string Name => Model.Name;
+         /// <summary> The provisioning state of the subnet resource. </summary>
         public ProvisioningState? ProvisioningState => Model.ProvisioningState;
+
+        /// <summary>  
+        /// A read-only string identifying the intention of use for this subnet based on
```
Comment (m-nash) [inline azure-proto-network/Placeholder/SubnetData.cs:24]: should be one line

File: azure-proto-network/Placeholder/SubnetData.cs [hunk 6]
```diff
@@ -5,77 +5,118 @@
...
+        /// <summary>An array of references to the external resources using subnet. </summary>
         public IList<ResourceNavigationLink> ResourceNavigationLinks => Model.ResourceNavigationLinks;
+        
+        /// <summary> Array of IpAllocation which reference this subnet. </summary>
         public IList<SubResource> IpAllocations
         {
             get => Model.IpAllocations;
             set => Model.IpAllocations = value;
         }
+        /// <summary>  Array of IP configuration profiles which reference this subnet. </summary>
```
Comment (m-nash) [inline azure-proto-network/Placeholder/SubnetData.cs:48]: missing newline

File: azure-proto-network/Placeholder/SubnetData.cs [hunk 7]
```diff
@@ -5,77 +5,118 @@
...
+        /// <summary> An array of references to private endpoints. </summary>
         public IList<PrivateEndpoint> PrivateEndpoints => Model.PrivateEndpoints;
+
+        /// <summary> An array of service endpoints. </summary>
         public IList<ServiceEndpointPropertiesFormat> ServiceEndpoints
         {
             get => Model.ServiceEndpoints;
             set => Model.ServiceEndpoints = value;
         }
+        /// <summary> Nat gateway associated with this subnet. </summary>
```
Comment (m-nash) [inline azure-proto-network/Placeholder/SubnetData.cs:70]: missing newline

File: azure-proto-network/Placeholder/SubnetData.cs [hunk 8]
```diff
@@ -5,77 +5,118 @@
...
+
+        /// <summary> Enable or Disable apply network policies on private link service in the subnet. </summary>
         public string PrivateLinkServiceNetworkPolicies
         {
             get => Model.PrivateLinkServiceNetworkPolicies;
             set => Model.PrivateLinkServiceNetworkPolicies = value;
         }
     }
-}
+}
```
Comment (m-nash) [inline azure-proto-network/Placeholder/SubnetData.cs:122]: missing newline at end of file

File: azure-proto-network/Subnet.cs [hunk 9]
```diff
@@ -12,4 +15,4 @@ public Subnet(AzureResourceManagerClientOptions SubnetOptions, SubnetData resour
 
         public SubnetData Data { get; private set; }
     }
-}
+}
```
Comment (m-nash) [inline azure-proto-network/Subnet.cs:18]: missing newline at end of file

File: azure-proto-network/SubnetContainer.cs [hunk 10]
```diff
@@ -91,4 +91,4 @@ public AsyncPageable<SubnetOperations> ListAsync(CancellationToken cancellationT
             return s => new Subnet(ClientOptions, new SubnetData(s, Location.Default));
         }
     }
-}
```
Comment (m-nash) [inline azure-proto-network/SubnetContainer.cs:94]: missing newline at end of file

File: azure-proto-network/SubnetOperations.cs [hunk 11]
```diff
@@ -67,4 +67,4 @@ public async override Task<ArmResponse<Subnet>> GetAsync(CancellationToken cance
                 });
         }
     }
-}
+}
```
Comment (m-nash) [inline azure-proto-network/SubnetOperations.cs:70]: missing newline at end of file

**Review by m-nash — CHANGES_REQUESTED** [review]

File: azure-proto-network/Placeholder/SubnetData.cs [hunk 12]
```diff
@@ -4,30 +4,21 @@
 
 namespace azure_proto_network
 {
-    /// <summary>
-    /// A class representing the subnet data model.
-    /// </summary>
+    /// <summary> A class representing the subnet data model. </summary>
```
Comment (m-nash) [inline azure-proto-network/Placeholder/SubnetData.cs:7]: this shouldn't be 1 line, the properties are copy paste from the existing code gen so they can be 1 line, but summary in general should be multi-line

File: azure-proto-network/Placeholder/SubnetData.cs [hunk 13]
```diff
@@ -4,30 +4,21 @@
...
     public class SubnetData : ProxyResource<Azure.ResourceManager.Network.Models.Subnet>
     {
-        /// <summary>
-        /// Initializes a new instance of the <see cref="SubnetData"/> class.
-        /// </summary>
+        /// <summary> Initializes a new instance of the <see cref="SubnetData"/> class. </summary>
```
Comment (m-nash) [inline azure-proto-network/Placeholder/SubnetData.cs:10]: this shouldn't be 1 line, the properties are copy paste from the existing code gen so they can be 1 line, but summary in general should be multi-line

File: azure-proto-network/Placeholder/SubnetData.cs (hunk 4 above)
Comment (m-nash) [reply azure-proto-network/Placeholder/SubnetData.cs:34]: still missing space

**Review by m-nash — CHANGES_REQUESTED** [review]

**Review by m-nash — APPROVED** [review]

**Pull request merged**